### DISCUSSION TITLE
IM server

### DISCUSSION THREAD

**Discussion by carr123** [nats-io/nats-server#2259] [Q&A] [answered]
Is nats-server suitable for serving as an IM server ?
many clients (say 10K) connect to a nats server cluster, each client subscribe an unique topic(userid as topic name).  and exchange messages between each other.
is nats-server work fine in this scenario ? 
thanks a lot.

**Comment by ripienaar** [comment, marked as answer]
Yes that connection and subscription count is quite small even for a single nats server.

**Reply by carr123** [reply]
thanks. is there any blog posts describing the addressing and discovering mechanism inside nats-server ?
it sounds amazing and magic that one message can find its subscribers across servers .

**Reply by derekcollison** [reply]
This may help a bit.
https://medium.com/hackernoon/want-a-super-fast-messaging-system-c19c9f8cf560

**Reply by carr123** [reply]
thanks a lot
